### PR TITLE
[Feature Highlight] Rotation fixes and animation improvement

### PR DESCRIPTION
**Pull request by samnm**
This PR replaces the mechanism for how feature highlights were reoriented upon rotation. Instead of using KVO on the frame of the highlighted view we now update on size class transition.

This change also makes the rotation animation look correct throughout the entire rotation.

### CODE REVIEW

File: components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.m [hunk 1]
```diff
@@ -193,7 +193,7 @@ - (void)setHighlightPoint:(CGPoint)highlightPoint {
   _highlightPoint = highlightPoint;
 
   [self setNeedsLayout];
-  [self layoutIfNeeded];
+//  [self layoutIfNeeded];
```
Comment (ianegordon) [inline components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.m]: Remove?

**Review by ianegordon — APPROVED** [review]
Please address comment before submit.

**Review by ianegordon — CHANGES_REQUESTED** [review]
There appear to be some issues with rotation of the main sample in catalog.

**Review by randallli — APPROVED** [review]

**Review by ianegordon — APPROVED** [review]
Rotation now works as expected.

**Pull request merged**